### PR TITLE
Implement needPermissions by currying restrict

### PR DESCRIPTION
**Pull request by DeltaEvo**

### CODE REVIEW

File: src/decorators/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import command from './command'
 import configurable from './configurable'
-import needPermissions from './needPermissions'
 import on from './on'
 import restrict from './restrict'
 
+export * from './restrict'
 export { command, configurable, needPermissions, on, restrict }
```
Comment (lightdiscord) [inline src/decorators/index.js:7]: Maybe remove `restrict` here (Line 7) ?

Comment (lightdiscord) [reply src/decorators/index.js:7]: Oh no, I didn't notice restrict was default export, my bad.

**Review by Sebastien-Ahkrin — DISMISSED** [review]

**Review by lightdiscord — DISMISSED** [review]

File: src/decorators/restrict.js [hunk 2]
```diff
@@ -9,3 +9,6 @@ export default function restrict(func) {
 		return descriptor
 	}
 }
+
+export const needPermissions = (permissions, ...args) =>
+	restrict(({ member }) => member && member.hasPermission(permissions, ...args))
```
Comment (cchudant) [inline src/decorators/restrict.js]: why don't you just do this
```js
export const needPermissions = (...args) =>
	restrict(({ member }) => member && member.hasPermission(...args))

Comment (DeltaEvo) [reply src/decorators/restrict.js]: Idk xD, I just copied the arguments in the original `needPermissions`

**Review by Sebastien-Ahkrin — APPROVED** [review]

**Review by lightdiscord — APPROVED** [review]

**Review by cchudant — APPROVED** [review]
oh yeah i love curry

**Pull request merged**